trame/state/core.py

```diff
@@ -164,4 +170,26 @@ def __getattr__(name):
 
     @staticmethod
     def __setattr__(name, value):
+        # Do not allow pre-existing attributes, such as update(), to be
+        # re-defined.
+        if name in State.__dict__:
+            msg = (
+                f"'{name}' is a special attribute on State that cannot be "
+                "re-assigned"
+            )
+            raise Exception(msg)
+
         update_state(name, value)
+
+    @staticmethod
+    def __getitem__(name):
+        return State.__getattr__(name)
+
+    @staticmethod
+    def __setitem__(name, value):
+        State.__setattr__(name, value)
+
+    @staticmethod
+    def update(d):
+        for k, v in d.items():
+            State.__setattr__(k, v)
```

Comment:
the method `update_state({})` manage that use case directly. There might be a batch handling when using the method directly but not sure. It might actually be the same. You may want to double check.

Reply:
I modified it to use `update_state()`